### ISSUE TITLE
Compress Log File Rollover

### ISSUE THREAD

**Issue by Knight-Yoshi** [yiisoft/yii2#13854]
Yii2 should have the option to compress rollover files.

**Comment by samdark** [comment]
Rollover files are the ones produced by file log target when primary file exceeds maximum size.

**Comment by Knight-Yoshi** [comment]
Yeah, and there should be an option to compress the files when they rollover to take up less disk space.

**Comment by samdark** [comment]
I think it makes sense to have it.

**Comment by letsjump** [comment]
As many operating systems has their own configurable _log file rotator_, we could write just write a widget that do the work, so Yii core doesn't grows too much.
If you are on a Debian SO with shell access, for example, you can attach a logrotate by editing `/etc/logrotate.conf`:
`/path/to/my/yiiApp/runtime/logs/*.log {
        weekly
        rotate 52
        delaycompress
        notifempty
        compress
        create 640 owner group
}
`
for more info on logrotate, [read its documentation](http://www.linuxcommand.org/man_pages/logrotate8.html)
